File: app/src/main/java/br/com/jonathanzanella/myexpenses/bill/BillInterface.java
package br.com.jonathanzanella.myexpenses.bill;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface BillInterface {
    @GET("bills")
    Call<List<Bill>> index(@Query("last-updated-at") long lastUpdatedAt);
    @POST("bills")
    Call<Bill> create(@Body Bill bill);
    @PUT("bills/{id}")
    Call<Bill> update(@Path("id") String serverId, @Body Bill bill);
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/bill/BillInterface.kt
package br.com.jonathanzanella.myexpenses.bill

import retrofit2.Call
import retrofit2.http.*

interface BillInterface {
    @GET("bills")
    fun index(@Query("last-updated-at") lastUpdatedAt: Long): Call<List<Bill>>

    @POST("bills")
    fun create(@Body bill: Bill): Call<Bill>

    @PUT("bills/{id}")
    fun update(@Path("id") serverId: String, @Body bill: Bill): Call<Bill>
}
